Amazon S3 Extension: Interceptors Configuration Fails to AWS Shipped Interceptors

**Describe the bug**
`quarkus.s3.interceptors=com.amazonaws.xray.interceptors.TracingInterceptor` results in a failure to configure the s3 client.

**Expected behavior**
AWS shipped S3 interceptors, like the XRay Tracing Interceptors, work with extension configuration model.

**Actual behavior**
(Describe the actual behavior clearly and concisely.)
```
Caused by: io.quarkus.builder.BuildException: 
Build failure: Build failed due to errors
        [error]: Build step io.quarkus.amazon.common.deployment.AmazonServicesClientsProcessor#setup threw an exception: io.quarkus.deployment.configuration.ConfigurationError: quarkus.s3.interceptors (Optional[[class com.amazonaws.xray.interceptors.TracingInterceptor]]) - must list only existing implementations of software.amazon.awssdk.core.interceptor.ExecutionInterceptor
        at io.quarkus.amazon.common.deployment.AmazonServicesClientsProcessor.lambda$setup$2(AmazonServicesClientsProcessor.java:73)
```

**To Reproduce**

Don't have a reproducer easily available. Should be able to generate a project with `quarkus-amazon-lambda-xray`, `quarkus-amazon-lambda` and `quarkus-amazon-s3` extensions, add the configuration below, and  create a junit where you inject an S3 client. Run the unit test.

Possible that you could also tweak https://github.com/quarkusio/quarkus/blob/master/integration-tests/amazon-services/src/main/resources/application.properties#L9-L15 and using the existing integration tests?

**Configuration**
```properties
quarkus.s3.interceptors=com.amazonaws.xray.interceptors.TracingInterceptor

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 
 - GraalVM version (if different from Java): 20.1.0-java8
 - Quarkus version or git rev: 1.8.3.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 3.6.3

**Additional context**
cc @marcinczeczko @gsmet @oztimpower This was precisely the sort of concern I had when I wrote https://github.com/quarkusio/quarkus/pull/4968#issuecomment-611186155. It's possible there is no xray extension yet, and this is the problem? Happy to test patches if that is helpful resolving this issue, but writing a new extension is probably more than I can commit. For now, I'll continue configuring the AWS SDK without the quarkus extensions.

--- a/extensions/amazon-services/common/deployment/src/main/java/io/quarkus/amazon/common/deployment/AmazonServicesClientsProcessor.java
+++ b/extensions/amazon-services/common/deployment/src/main/java/io/quarkus/amazon/common/deployment/AmazonServicesClientsProcessor.java
@@ -27,6 +27,7 @@ import software.amazon.awssdk.http.async.SdkAsyncHttpService;
 
 public class AmazonServicesClientsProcessor {
     public static final String AWS_SDK_APPLICATION_ARCHIVE_MARKERS = "software/amazon/awssdk";
+    public static final String AWS_SDK_XRAY_ARCHIVE_MARKER = "com/amazonaws/xray";
 
     private static final String APACHE_HTTP_SERVICE = "software.amazon.awssdk.http.apache.ApacheSdkHttpService";
     private static final String NETTY_HTTP_SERVICE = "software.amazon.awssdk.http.nio.netty.NettySdkAsyncHttpService";
@@ -41,8 +42,9 @@ public class AmazonServicesClientsProcessor {
     }
 
     @BuildStep
-    AdditionalApplicationArchiveMarkerBuildItem awsAppArchiveMarkers() {
-        return new AdditionalApplicationArchiveMarkerBuildItem(AWS_SDK_APPLICATION_ARCHIVE_MARKERS);
+    void awsAppArchiveMarkers(BuildProducer<AdditionalApplicationArchiveMarkerBuildItem> archiveMarker) {
+        archiveMarker.produce(new AdditionalApplicationArchiveMarkerBuildItem(AWS_SDK_APPLICATION_ARCHIVE_MARKERS));
+        archiveMarker.produce(new AdditionalApplicationArchiveMarkerBuildItem(AWS_SDK_XRAY_ARCHIVE_MARKER));
     }
 
     @BuildStep

